src/geff/core_io/_utils.py

```diff
@@ -172,3 +178,71 @@ def setup_zarr_group(store: StoreLike, zarr_format: Literal[2, 3] = 2) -> zarr.G
         return zarr.open_group(store, mode="a", zarr_format=zarr_format)
     else:
         return zarr.open_group(store, mode="a")
+
+
+def _get_common_type_dims(arr_seq: Sequence[ArrayLike | None]) -> tuple[np.dtype, int]:
+    ndim = None
+    dtype = None
+
+    for arr in arr_seq:
+        if arr is None:
+            continue
+        element = np.asarray(arr)
+        if ndim is None:
+            ndim = element.ndim
+            dtype = element.dtype
+        else:
+            ndim = max(element.ndim, ndim)
+            if np.can_cast(dtype, element.dtype):
+                dtype = np.promote_types(dtype, element.dtype)
+            else:
+                raise ValueError(
+                    "All elements must have compatible dtypes. Cannot"
+                    f"cast {dtype} and {element.dtype}."
+                )
+
+    if dtype is None or ndim is None:
+        warnings.warn(
+            "Variable length property sequence does not have any valid elements - "
+            "using ndim=1 and dtype=int64",
+            stacklevel=2,
+        )
+        dtype = np.dtype("int64")
+        ndim = 1
+    return dtype, ndim
+
+
+def construct_var_len_props(arr_seq: Sequence[ArrayLike | None]) -> PropDictNpArray:
+    """Converts a sequence of array like and None objects into a geff._typing.PropDictNpArray
+
+    Creates a missing array with the indices of the None objects. Converts each element of
+    the sequence into a numpy array. Prepends dummy dimensions to each element to ensure
+    all elements have the same number of dimensions. Casts all arrays into a common dtype
+    (if there is one). Turns the sequence into a numpy array with dtype object.
+
+    Args:
+        arr_seq (Sequence[ArrayLike | None]): A sequence of properties, with one entry
+            per node or edge. Missing values are indicated by None entries.
+
+    Returns:
+        PropDictNpArray: A standardized version of the input properties where all entries
+        are numpy arrays contained in an object array.
```

Comment:
```suggestion
            are numpy arrays contained in an object array.
```
Fix docstring rendering

Reply:
I just subconsciously fixed this when I was reading through the file to add the other docstring 😆 My brain knows what sphinx wants